google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1beta2/st/ITBigQueryStorageLongRunningWriteTest.java

```diff
@@ -0,0 +1,307 @@
+                  "42", "43", "44", "45", "46", "47", "48", "49", "50", "51", "52", "53", "54",
+                  "55", "56", "57", "58", "59", "60", "61", "62", "63", "64", "65", "66", "67",
+                  "68", "69", "70", "71", "72", "73", "74", "75", "76", "77", "78", "79", "80",
+                  "81", "82", "83", "84", "85", "86", "87", "88", "89", "90", "91", "92", "93",
+                  "94", "95", "96", "97", "98", "99", "100"
+                }));
+        object.put("test_datetime", String.valueOf(LocalDateTime.now()));
+        object.put(
+            "test_bools",
+            new JSONArray(
+                new boolean[] {
+                  false, true, false, true, false, true, false, true, false, true, false, true,
+                  false, true, false, true, false, true, true, false, true, false, true, false,
+                  true, false, true, false, true, false, true, true, false, true, false, true,
+                  false, true, false, true, false, true, false, true, true, false, true, false,
+                  true, false, true, false, true, false, true, false, true, true, false, true,
+                  false, true, false, true, false, true, false, true, false, true, true, false,
+                  true, false, true, false, true, false, true, false, true, false, true, true,
+                  false, true, false, true, false, true, false, true, false, true, false, true,
+                  true, false, true, false, true, false, true, false, true, false, true, false,
+                  true, true, false, true, false, true, false, true, false, true, false, true,
+                  false, true,
+                }));
+        JSONObject sub = new JSONObject();
+        sub.put("sub_bool", true);
+        sub.put("sub_int", 12);
+        sub.put("sub_string", "Test Test Test");
+        object.put("test_subs", new JSONArray(new JSONObject[] {sub, sub, sub, sub, sub, sub}));
+        break;
+      default:
+        break;
+    }
+    return object;
+  }
+
+  @BeforeClass
+  public static void beforeClass() throws IOException {
+    parentProjectId = String.format("projects/%s", ServiceOptions.getDefaultProjectId());
+
+    client = BigQueryWriteClient.create();
+    RemoteBigQueryHelper bigqueryHelper = RemoteBigQueryHelper.create();
+    bigquery = bigqueryHelper.getOptions().getService();
+    dataset = RemoteBigQueryHelper.generateDatasetName();
+    DatasetInfo datasetInfo =
+        DatasetInfo.newBuilder(/* datasetId = */ dataset).setDescription(DESCRIPTION).build();
+    bigquery.create(datasetInfo);
+  }
+
+  @AfterClass
+  public static void afterClass() {
+    if (client != null) {
+      client.close();
+    }
+    if (bigquery != null && dataset != null) {
+      RemoteBigQueryHelper.forceDelete(bigquery, dataset);
+      LOG.info("Deleted test dataset: " + dataset);
+    }
+  }
+
+  @Test
+  public void testDefaultStream()
```

Comment:
Split this into two tests:
testDefaultStreamSimpleSchema
testDefaultStreamComplexSchema

They should be able to be run independently, plus this adds a lot of unnecessary complexity (e.g. code size, code complexity, shared variables/resetting metrics in between).  It may also be worth considering whether you can reuse some of the code between them since it's very similar.

As I mentioned above, it'll probably be cleaner to add the complex-schema part in another pr.  Smaller/more focused changes tend to get reviewed faster and more accurately than large changes.

Reply:
yeah that makes sense, I'll separate that out and try to pull repeat code out into functions. For now I think I'll just pull the complex part out of this PR and put it into another one.